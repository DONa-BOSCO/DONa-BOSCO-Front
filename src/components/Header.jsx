import React from 'react'
import Button from 'react-bootstrap/Button';
import Container from 'react-bootstrap/Container';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import { Link } from 'react-router-dom';

const Header = () => {
  return (
    <Navbar collapseOnSelect className="Shadow" expand="lg" bg="light" variant="light">
      <Container>
      <Navbar.Brand>
        <Link to= "/" >

            <img
              src="/logo.png"
              width="140"
              height="70"
              alt="React Bootstrap logo"
            /></Link>
          </Navbar.Brand>
          
        <Navbar.Toggle aria-controls="responsive-navbar-nav" />
        <Navbar.Collapse id="responsive-navbar-nav">
          <Nav className="justify-content-end flex-grow-1 me-5">
            <Nav.Link href="/Contact">Contacto</Nav.Link>
          </Nav>

          <Nav>
          <Link to="/Login">
          <Button className="me-4 btn btn-custom">Iniciar Sesi√≥n</Button>
          </Link>
          </Nav>
          <Nav>
            <Link to="/Join">
          <Button className="me-4 btn btn-custom">Registrar</Button>
          </Link>
          </Nav>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  )
}
export default Header