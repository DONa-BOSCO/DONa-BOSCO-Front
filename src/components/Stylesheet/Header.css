.nav-btn {
    margin-right: 1rem;
  }
  
  @media (max-width: 991px) {
    .nav-btn {
      margin-right: 0;
      margin-bottom: 0.5rem;
    }
  }
  .nav-link-contact::after {
    content: '';
    display: block;
    height: 2px;
    width: 0;
    background-color: #dc3545;
    transition: width 0.3s ease-in-out;
  }
  .nav-link-contact:hover::after {
    width: 100%;
  }
  .Navbar .Nav .nav-link:hover {
    font-weight: bold;
    font-size: 1.2rem;
    transition: all 0.2s ease-in-out;
  }
  