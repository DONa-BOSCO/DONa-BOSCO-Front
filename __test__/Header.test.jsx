import React from 'react';
import Nav from 'react-bootstrap/Nav';
import { Link } from "react-router-dom";
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import Header from '../src/components/Header'

describe('Show navbar contact', () => {
    it('renders text correctly', () => {
        const sut = render(
            <Router>
                <Header />
            </Router>
        );
        expect(sut).toMatchSnapshot();
    });

    it('Nav component renders correctly', () => {
        const component = render(
            <Router>
                <Nav>
                    <Link to='/login'></Link>
                    <Link to='/join'></Link>
                </Nav>
            </Router>
        );
        expect(component).toMatchSnapshot();
    });

      
});
