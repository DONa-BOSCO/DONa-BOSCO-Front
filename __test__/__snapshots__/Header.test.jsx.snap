// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Show navbar contact > Nav component renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="nav"
      >
        <a
          href="/login"
        />
        <a
          href="/join"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="nav"
    >
      <a
        href="/login"
      />
      <a
        href="/join"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Show navbar contact > renders text correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="Shadow  navbar navbar-expand-lg navbar-light bg-light"
        style="box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.25);"
      >
        <div
          class="container"
        >
          <span
            class="navbar-brand"
          >
            <a
              href="/"
            >
              <img
                alt="Dona Bosco logo"
                height="60"
                src="/logo.png"
                width="120"
              />
            </a>
          </span>
          <button
            aria-controls="responsive-navbar-nav"
            aria-label="Toggle navigation"
            class="navbar-toggler collapsed"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="navbar-collapse collapse"
            id="responsive-navbar-nav"
          >
            <div
              class="justify-content-end flex-grow-1 me-5 mt-4 navbar-nav"
            >
              <a
                class="nav-link-contact nav-link"
                data-rr-ui-event-key="/Contact"
                href="/Contact"
              >
                <h5>
                  Contacto
                </h5>
              </a>
            </div>
            <div
              class="navbar-nav"
            >
              <a
                href="/Login"
              >
                <button
                  class="me-4 btn btn-custom nav-btn btn btn-primary"
                  style="border-color: #dc3545;"
                  type="button"
                >
                  Iniciar Sesión
                </button>
              </a>
            </div>
            <div
              class="navbar-nav"
            >
              <a
                href="/Join"
              >
                <button
                  class="me-4 btn btn-custom btn btn-primary"
                  style="border-color: #dc3545;"
                  type="button"
                >
                  Registrar
                </button>
              </a>
            </div>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="Shadow  navbar navbar-expand-lg navbar-light bg-light"
      style="box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.25);"
    >
      <div
        class="container"
      >
        <span
          class="navbar-brand"
        >
          <a
            href="/"
          >
            <img
              alt="Dona Bosco logo"
              height="60"
              src="/logo.png"
              width="120"
            />
          </a>
        </span>
        <button
          aria-controls="responsive-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="responsive-navbar-nav"
        >
          <div
            class="justify-content-end flex-grow-1 me-5 mt-4 navbar-nav"
          >
            <a
              class="nav-link-contact nav-link"
              data-rr-ui-event-key="/Contact"
              href="/Contact"
            >
              <h5>
                Contacto
              </h5>
            </a>
          </div>
          <div
            class="navbar-nav"
          >
            <a
              href="/Login"
            >
              <button
                class="me-4 btn btn-custom nav-btn btn btn-primary"
                style="border-color: #dc3545;"
                type="button"
              >
                Iniciar Sesión
              </button>
            </a>
          </div>
          <div
            class="navbar-nav"
          >
            <a
              href="/Join"
            >
              <button
                class="me-4 btn btn-custom btn btn-primary"
                style="border-color: #dc3545;"
                type="button"
              >
                Registrar
              </button>
            </a>
          </div>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
